### PR TITLE
Feature/decorator

### PR DESCRIPTION
**Pull request by Wikitelmex**
- add corrector decorator, that composite from Person class to decorate an instance of this class, modifying name to capitalize it and trim to max 10 characters.

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# corrector_object1 = Corrector.new(person_object1)
+# person_object1.name # => "John"
+#
+# Date: Dec, 22 2021
+
+require './person'
+
+class Corrector
+  def initialize(person: Person)
+    person.name = correct_name(person.name)
```
Comment (descholar-ceo) [inline corrector.rb]: - [x] There should be an instance of Corrector set in the Person class on initialization
- [x] We should have `validate_name` method in the `Person` class.

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Wikitelmex , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: corrector.rb (hunk 1 above)
Comment (Wikitelmex) [reply corrector.rb]: Ohh, you are right, I didn't look in-depth to that requirements, anyway, those requirements, are not usual to a decorator pattern, and make you violate the Solid principles.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi @Wikitelmex, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍
Congratulations! 🎉 

Great job working on the changes requested

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**